### PR TITLE
Implement ListPrioCommand Class and PriorityMatchesPredicate

### LINKED ISSUE #89: Implement ListPrioCommand class

### PR DESCRIPTION
**Pull request by PresleyChew**
fixes #89 

Test cases for both aforementioned class are implemented as well. Will implement ListPrioCommandParser next.

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/ListPrioCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    @Test
+    public void execute_nonePriority_success() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 1);
+        PriorityMatchesPredicate predicate = new PriorityMatchesPredicate(noPriority);
+        ListPrioCommand command = new ListPrioCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(GEORGE), model.getFilteredPersonList());
+    }
```
Comment (yong-zhuo) [inline src/test/java/seedu/address/logic/commands/ListPrioCommandTest.java:101]: Test cases are comprehensive and checks every possible scenario for `Priority` 👍🏻

**Review by yong-zhuo — APPROVED** [review]
Overall LGTM, test cases are done well and there is no coding violations. 👍🏻

File: src/main/java/seedu/address/logic/commands/ListPrioCommand.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            + "Parameters: Priority\n"
+            + "Example: " + COMMAND_WORD + " HIGH";
+
+    private final PriorityMatchesPredicate predicate;
+
+    /**
+     * Constructs a ListPrioCommand to filter and list patients by their Priority level.
+     *
+     * @param predicate The predicate to filter patients by their Priority level.
+     */
```
Comment (hweinian) [inline src/main/java/seedu/address/logic/commands/ListPrioCommand.java:29]: LGTM

File: src/test/java/seedu/address/logic/commands/ListPrioCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import seedu.address.model.person.PriorityMatchesPredicate;
+
+public class ListPrioCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private Priority noPriority = new Priority();
+    private Priority lowPriority = new Priority("LOW");
+    private Priority mediumPriority = new Priority("MEDIUM");
+    private Priority highPriority = new Priority("HIGH");
```
Comment (hweinian) [inline src/test/java/seedu/address/logic/commands/ListPrioCommandTest.java:34]: extensive testing

**Review by hweinian — COMMENTED** [review]
LGTM

**Pull request merged**